### ISSUE TITLE
LeftCurly: False Negative  when multiple blocks start on a single line

### ISSUE THREAD

**Issue by leemeii** [checkstyle/checkstyle#19162]
I have read check documentation: https://checkstyle.org/checks/blocks/leftcurly.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac LeftCurlyTest.java
# [[SUCCESSFULLY COMPILED]]

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN" "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="LeftCurly">
      <property name="option" value="eol"/>
    </module>
  </module>
</module>

/var/tmp $ cat LeftCurlyTest.java
/** Javadoc */
public class LeftCurlyTest {
    // FN Case: CLASS_DEF curly should be at the end of the line, but it's followed by CTOR_DEF.
    // Checkstyle stays silent because the line ends with another '{' from CTOR_DEF.
    public class FNCase { public FNCase() {
        System.out.println("No violation reported for the first brace.");
    } }

    // TP Case: Reported correctly as the line does not end with a brace.
    public void tpMethod() { int x = 0;
        System.out.println(x);
    }
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-13.3.0-all.jar -c config.xml LeftCurlyTest.java
Starting audit...
[WARN] LeftCurlyTest.java:13:28: '{' at column 28 should have line break after. [LeftCurly]
Audit done.
```

**Describe what you expect in detail.**
Checkstyle should report a LeftCurly violation for the brace at column 25 (the CLASS_DEF brace) on line 6.

---

**Actual Behavior:**
On line 6: public class FNCase { public FNCase() {, there are two LCURLY tokens. The first one belongs to CLASS_DEF, and the second belongs to CTOR_DEF. The first brace is clearly NOT at the end of the line. However, Checkstyle fails to report a violation for the first brace.

**Comment by abhii-os** [comment]
Issue: False Negative in LeftCurly (option=eol) when multiple braces exist on one line
Description
The LeftCurly check with the option property set to eol fails to report a violation when a curly brace (e.g., for CLASS_DEF) is followed by more code (e.g., a CTOR_DEF) on the same line, provided the line eventually ends with another curly brace.

The check appears to validate that a brace exists at the end of the line, rather than ensuring each brace is followed immediately by a newline.

Minimal Working Example
config.xml

`XML

<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN" "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="LeftCurly">
      <property name="option" value="eol"/>
    </module>
  </module>
</module>
LeftCurlyTest.java

Java

/** Javadoc */
public class LeftCurlyTest {
    // FN Case: CLASS_DEF curly should be at the end of the line.
    // Checkstyle stays silent because the line ends with another '{' from CTOR_DEF.
    public class FNCase { public FNCase() {
        System.out.println("No violation reported.");
    } }
}
Execution Command
Bash

java -jar checkstyle-13.3.0-all.jar -c config.xml LeftCurlyTest.java
Expected Result
The audit should report two violations for line 5:

Column 25: The { for CLASS_DEF is not at the end of the line.

Column 43: The { for CTOR_DEF is not at the end of the line.

Actual Result
Plaintext

Starting audit...
Audit done.
(No violations reported for the FNCase line)

Why this happens
The LeftCurly check iterates through tokens. When it reaches the CLASS_DEF brace, it likely checks the remaining characters on the line. If it sees another LCURLY later on, or if the logic specifically looks for the last character of the line to be a brace, it erroneously satisfies the eol requirement for the first brace.

**Comment by Feliks-WR** [comment]
I'm on it

Test added in my fork
